### PR TITLE
E2E operator conversion webhook

### LINKED ISSUE #299: [e2e test operator] conversion webhook
The e2e test operator should have conversion webhooks registered for it. For additional details see epic's root issue, #296 

Acceptance criteria: 

* [x] some operator CRDs should come in two different version
* [x] the version may be copy-compatible (that is having the same schema, but different `apiGroup`) though it's better if they have differing incompatible schemas. 
* [x] The conversion should work flawlessly and  without errors (minus the possible network errors, but that's out of scope this PR)

### PR DESCRIPTION
**Pull request by aborilov**
**What this PR does / why we need it**:
The e2e test operator should have conversion webhooks registered for it.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #299 

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/fake/v1/db_types.go [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+// OperationFlagType represents a enable/disable flag
+type OperationFlagType string
+
+func (o OperationFlagType) Enabled() bool {
+	return o == OperationFlagEnabled
+}
+
+// Values of OperationFlagType.
+const (
+	OperationFlagEnabled  OperationFlagType = "enabled"
```
Comment (jiachengxu) [inline pkg/apis/fake/v1/db_types.go:51]: As we discussed in the webhook PR, can you please use strings with initial capitalized letters?

File: pkg/apis/fake/v1/db_types.go [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+	OperationFlagDisabled OperationFlagType = "disabled"
+)
+
+// Config defines the e2e tests params
+type Config struct {
+	// ReadyAfterSeconds represents duration after which operator will mark DB as Ready
+	// +kubebuilder:default:=0
+	ReadyAfterSeconds int `json:"readyAfterSeconds,omitempty"`
+	// DeletionAfterSeconds represents duration after which operator will remove finalizer
+	// +kubebuilder:default:=0
```
Comment (jiachengxu) [inline pkg/apis/fake/v1/db_types.go:61]: No need to default the value to 0

File: pkg/fakeoperator/internal/controllers/db_controller.go [hunk 3]
```diff
@@ -58,7 +58,7 @@ func (r *DBReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		}
 		cond, _ := db.Status.GetCondition(fakev1alpha1.DBReady)
 		if time.Since(cond.LastTransitionTime.Time) < time.Duration(db.Spec.Config.DeletionAfterSeconds)*time.Second {
-			return ctrl.Result{RequeueAfter: time.Second * time.Duration(db.Spec.Config.DeletionAfterSeconds)}, nil
+			return ctrl.Result{RequeueAfter: time.Second*time.Duration(db.Spec.Config.DeletionAfterSeconds) + 10*time.Millisecond}, nil
```
Comment (jiachengxu) [inline pkg/fakeoperator/internal/controllers/db_controller.go:61]: why is this changed?

File: pkg/apis/fake/v1/db_types.go (hunk 1 above)
Comment (aborilov) [reply pkg/apis/fake/v1/db_types.go:51]: done

File: pkg/apis/fake/v1/db_types.go (hunk 2 above)
Comment (aborilov) [reply pkg/apis/fake/v1/db_types.go:61]: done

File: pkg/fakeoperator/internal/controllers/db_controller.go (hunk 3 above)
Comment (aborilov) [reply pkg/fakeoperator/internal/controllers/db_controller.go:61]: I had a problem when I can delete an instance, I thought it's possible when we have `DeletionAfterSeconds=0` and deletion happen in the same second as we change the condition to terminating, but now I see that it's irrelevant.
so I will revert this change
